pycytominer_transform/convert.py

```diff
@@ -0,0 +1,582 @@
+) -> Dict[str, List[Dict[str, Any]]]:
+    """
+    Filter source filepaths based on provided target_datatype
+
+    Args:
+      records: Dict[str, List[Dict[str, Any]]]
+        Grouped datasets of files which will be used by other functions.
+      target_datatype: str
+        Target datatype to use for filtering the dataset.
+
+    Returns:
+      Dict[str, List[Dict[str, Any]]]
+        Data structure which groups related files based on the targets.
+    """
+
+    return {
+        filegroup: [
+            file
+            for file in files
+            # ensure the filesize is greater than 0
+            if file["source_path"].stat().st_size > 0
+            # ensure the datatype matches the target datatype
+            and file["source_path"].suffix == f".{target_datatype}"
+        ]
+        for filegroup, files in records.items()
+    }
+
+
+@flow
+def gather_records(
+    source_path: str,
+    source_datatype: Optional[str] = None,
+    targets: Optional[List[str]] = None,
+    **kwargs,
+) -> Dict[str, List[Dict[str, Any]]]:
+
+    """
+    Flow for gathering records for conversion
+
+    Args:
+      source_path: str:
+        Where to gather file-based data from.
+      source_datatype: Optional[str]:  (Default value = None)
+        The source datatype (extension) to use for reading the tables.
+      targets: Optional[List[str]]:  (Default value = None)
+        The source file names to target within the provided path.
+
+    Returns:
+      Dict[str, List[Dict[str, Any]]]
+        Data structure which groups related files based on the targets.
+    """
+
+    source_path = build_path(path=source_path, **kwargs)
+
+    # gather filepaths which will be used as the basis for this work
+    records = get_source_filepaths(path=source_path, targets=targets)
+
+    # infer or validate the source datatype based on source filepaths
+    source_datatype = infer_source_datatype(
+        records=records, target_datatype=source_datatype
+    )
+
+    # filter source filepaths to inferred or targeted datatype
+    records = filter_source_filepaths(records=records, target_datatype=source_datatype)
+
+    return records
+
+
+@task
+def read_file(record: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Read csv file from record.
+
+    Args:
+      record: Dict[str, Any]:
+        Data containing filepath to csv file
+
+    Returns:
+      Dict[str, Any]
+        Updated dictionary with CSV data in-memory
+    """
+
+    if AnyPath(record["source_path"]).suffix == ".csv":  # pylint: disable=no-member
+        # define invalid row handler for rows which may be
+        # somehow erroneous. See below for more details:
+        # https://arrow.apache.org/docs/python/generated/pyarrow.csv.ParseOptions.html#pyarrow-csv-parseoptions
+        def skip_erroneous_colcount(row):
+            if row.actual_columns != row.expected_columns:
+                return "skip"
+            return "error"
```

Comment:
This may have been a pylint find and change ([R1705: no-else-return](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html)), I'll try out this change and see what occurs. Not a big deal if we need a one-off exception, but hope to keep this repo in alignment with those standards.

Reply:
Pylint turned out to be okay with this change, added it into a later commit. Thanks again! 👍